src/AutoComplete/StyledAutoComplete.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import ReactSelect from 'react-select';
+import ReactSelectAsync from 'react-select/async';
+import ReactSelectCreatable from 'react-select/creatable';
+import PropTypes from 'prop-types';
+import { withTheme } from 'emotion-theming';
+
+import StyledClearIndicator from './StyledClearIndicator';
+import StyledMultiValue from './StyledMultiValue';
+import reactSelectThemeOverrides from './reactSelectThemeOverrides';
+
+const addStyles = styles => provided => ({ ...provided, ...styles });
+
+const components = {
+  MultiValue: StyledMultiValue,
+  ClearIndicator: StyledClearIndicator
+};
+
+const getReactSelectComponent = variant => {
+  switch (variant) {
+    case 'creatable':
+      return ReactSelectCreatable;
+    case 'async':
+      return ReactSelectAsync;
+    default:
+      return ReactSelect;
+  }
+};
+
+const getStyles = ({ radii, shadows, brandFont, space }) => ({
+  container: addStyles({
+    fontFamily: brandFont,
+    marginBottom: space.smallest,
+    marginTop: space.smallest
+  }),
+  dropdownIndicator: addStyles({ display: 'none' }),
+  control: addStyles({ boxShadow: 'none' }),
+  indicatorSeparator: addStyles({ display: 'none' }),
+  menu: addStyles({
+    borderRadius: radii.large,
+    boxShadow: shadows.elevation2
+  }),
+  valueContainer: addStyles({ padding: `0 ${space.smaller}` })
+});
+
+const reactSelectTheme = providedTheme => ({
+  ...providedTheme,
+  ...reactSelectThemeOverrides
+});
+
+const StyledAutoComplete = ({ theme, variant, ...props }) => {
+  const SelectComponent = getReactSelectComponent(variant);
+  const reactSelectProps = {
+    components,
+    styles: getStyles(theme),
+    theme: reactSelectTheme,
+    ...props
+  };
+
+  return <SelectComponent {...{ ...reactSelectProps, ...props }} />;
+};
+
+StyledAutoComplete.propTypes = {
+  theme: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
+  variant: PropTypes.oneOf(['async', 'creatable', 'default', undefined])
+};
+
+StyledAutoComplete.defaultProps = {
+  variant: undefined
```

Comment:
Silly question, why is undefined the default prop for variant as opposed to 'default'?

Reply:
I went back and forth on this... default and undefined yield the same result. I thought it might be helpful to supply either nothing or the string default, especially in regards to switch statements or other conditionals. This was useful in the tests and example usages... that being said it might be a bit contrived.

Reply:
It feels a bit confusing to see both undefined and default. When I see a variant named 'default', I would assume that would be the default prop as opposed to undefined.